blog-tech/2022-10-20-k3s.md

```diff
@@ -0,0 +1,417 @@
+
+   <img src="/blog-221020/登入cloud.png" width="70%" />
+   
+   b. Add device
+
+   <img src="/blog-221020/添加设备.png" width="70%" />
+
+   c. Select Private Protocol --> Select Hikvision in the drop-down menu, then click Next
+
+   <img src="/blog-221020/选择协议.png" width="70%" />
+
+     
+   d. Add basic information about the device, such as device name, manufacturer and model:\
+
+   <img src="/blog-221020/添加基本信息.png" width="70%" />
+   
+   e. Fill in the IP address, RTSP username and password
+
+   <img src="/blog-221020/IP地址.png" width="60%" />
+   
+   f. Click Access and go to Add Interface
+
+   <img src="/blog-221020/前往添加端口.png" width="60%" />
+  
+   
+   g. Finally click Upload and ***Shifu Cloud*** will automatically generate the YAML file for ***deviceShifu*** (the digital twin)
+
+   <img src="/blog-221020/自动生成Yaml文件.png" width="60%" />
+   
+   h. Copy the command and run it on the master node of K3s
+   <img src="/blog-221020/复制命令.png" width="60%" />
+   
+   i. The output should be as follows:
+      ```
+      root@localhost:~# kubectl apply -f https://******.com/*****/******.yaml
+      configmap/deviceshifu-devicehikvision-configmap created
+      service/deviceshifu-devicehikvision-service created
+      deployment.apps/deviceshifu-devicehikvision-deployment created
+      edgedevice.shifu.edgenesis.io/edgedevice-devicehikvision created
+      root@localhost:~# 
+      ```
+
+      - What happens behind the scene
+      ***Shifu Cloud*** automatically generates the digital twin's YAML files (Edge Device, ConfigMap, Deployment and Service) with some basic information filled in by the user, if you want to go deeper, go to [github](https://github.com/Edgenesis/shifu/tree/main/examples/rtspDeviceShifu) to see the deployment specific files
+      
+   j. ***Shifu Cloud*** does not support adding nodeSelector for now (stay tuned), the device twin will be deployed on the master node by default, we need to update the `spec` in the deployment file to deploy the Pod in the worker node:
+
+    - We get the name of the current deployment file with the command
+         ```
+         root@localhost:~# kubectl get deployment -n deviceshifu
+         NAME READY UP-TO-DATE AVAILABLE AGE
+         deviceshifu-devicehikvision-deployment 0/1 1 0 16m
+         ```
+      
+    - Then edit the deployment with the ``kubectl edit deployment -n deviceshifu deviceshifu-devicehikvision-deployment`` command, add the following two lines and save it:
+      ```
+         ......
+           nodeSelector:                                                                                                                            
+           type: worker
+         ......
```

Comment:
need to revert the indentation changes in this section, we want to indent the outer ``` only